src/librustc_privacy/lib.rs

```diff
@@ -911,7 +911,13 @@ impl<'a, 'tcx: 'a> SearchInterfaceForPrivateItemsVisitor<'a, 'tcx> {
     }
 
     fn predicates(&mut self) -> &mut Self {
-        self.tcx.predicates_of(self.item_def_id).visit_with(self);
+        let predicates = self.tcx.predicates_of(self.item_def_id);
+        for predicate in &predicates.predicates {
+            if let &ty::Predicate::Trait(/* Trait(poly_predicate)*/poly_predicate) = predicate {
+                self.visit_trait_ref(poly_predicate.skip_binder().trait_ref);
+            }
+        }
+        predicates.visit_with(self);
```

Comment:
I'd call this on every predicate in the loop, otherwise the order might look weird.